packages/dynamic-import/dynamic-versions.js

```diff
@@ -30,3 +30,55 @@ exports.get = function (id) {
 
   return version;
 };
+
+function getFlatModuleArray(tree) {
+  var parts = [""];
+  var result = [];
+
+  function walk(t) {
+    if (t && typeof t === "object") {
+      Object.keys(t).forEach(function (key) {
+        parts.push(key);
+        walk(t[key]);
+        parts.pop();
+      });
+    } else if (typeof t === "string") {
+      result.push(parts.join("/"));
+    }
+  }
+
+  walk(tree);
+
+  return result;
+}
+
+// If Package.appcache is loaded, preload additional modules after
+// the core bundle has been loaded.
+function precacheOnLoad(event) {
+  // Check inside onload to make sure Package.appcache has had a
+  // chance to become available.
+  if (!Package.appcache) return;
+
+  // prefetch in chunks to reduce overhead
+  // If we call module.prefetch(id) multiple times in the same tick of
+  // the event loop, all those modules will be fetched in one request.
+  function prefetchInChunks(modules, amount) {
+    Promise.all(modules.splice(0, amount).map(function (id) {
+      return module.prefetch(id);
+    })).then(function () {
+      if (modules.length > 0) {
+        prefetchInChunks(modules, amount);
+      }
+    })
+  }
+
+  // Get a flat array of modules and start prefetching.
+  prefetchInChunks(getFlatModuleArray(versions), 20);
+}
+
+// Use window.onload to only prefetch after the main bundle has loaded
+if (global.addEventListener) {
+  global.addEventListener('load', precacheOnLoad, false);
+} else if (global.attachEvent) {
+  global.attachEvent('load', precacheOnLoad);
```

Comment:
This should be `onload` instead of `load` (but only for `global.attachEvent`). Thanks, IE.

Reply:
Ooph! Good catch.